src/vm/mod.rs

```diff
@@ -1,3 +1,115 @@
+    ///
+    /// println!("{:#?}", caps);
+    /// ```
+    pub fn get_capabilities(&self) -> Result<TDXCapabilities, kvm_ioctls::Error> {
+        // configure the kernel facing capabilities structure
+        let cpuid_configs: [CpuidConfig; NR_CPUID_CONFIGS] = [Default::default(); NR_CPUID_CONFIGS];
+
+        let caps = Capabilities {
+            attrs_fixed0: 0,
+            attrs_fixed1: 0,
+            xfam_fixed0: 0,
+            xfam_fixed1: 0,
+            supported_gpaw: 0,
+            padding: 0,
+            reserved: [0; 251],
+
+            nr_cpuid_configs: NR_CPUID_CONFIGS as u32,
+            cpuid_configs,
+        };
+
+        // configure the kernel facing TDX command that will be wrapped by the
+        //  MEMORY_ENCRYPT_OP ioctl
+        let mut cmd = Cmd {
+            id: CmdId::KvmTDXCapabilities as u32,
+            flags: 0,
+            data: &caps as *const Capabilities as _,
+            error: 0,
+            unused: 0,
+        };
+
+        unsafe { self.0.encrypt_op(&mut cmd)? };
+
+        // MEMORY_ENCRYPT_OP ioctl was a success so we can return the TDX
+        // capabilities in a user-friendly way
+        Ok(TDXCapabilities {
+            fixed_attrs: FixedAttrs {
+                fixed0: caps.attrs_fixed0,
+                fixed1: caps.attrs_fixed1,
+            },
+            fixed_xfam: FixedXFAM {
+                fixed0: caps.xfam_fixed0,
+                fixed1: caps.xfam_fixed1,
+            },
+            supported_gpaw: caps.supported_gpaw,
+            padding: caps.padding,
+            _reserved: caps.reserved,
+            cpuid_configuration: CpuidConfiguration {
+                nr_cpuid_configs: NR_CPUID_CONFIGS as u32,
+                cpuid_configs: caps.cpuid_configs,
+            },
+        })
+    }
+}
+
+/// Reflects the Intel TDX module capabilities and configuration and CPU
+/// capabilities
+#[derive(Debug)]
+pub struct FixedAttrs {
+    pub fixed0: u64,
+    pub fixed1: u64,
+}
+
+/// Determines the set of extended features available for use by the guest TD
+#[derive(Debug)]
+pub struct FixedXFAM {
+    pub fixed0: u64,
+    pub fixed1: u64,
+}
+
+/// Enumeration of the CPUID leaves/sub-leaves that contains bit fields whose
+/// virtualization by the Intel TDX module is either directly configurable
+/// (CONFIG_DIRECT) or bits that the host VMM may allow to be 1 (ALLOW_DIRECT)
+/// and their native value, as returned by the CPU, is 1
+#[derive(Debug)]
+pub struct CpuidConfiguration {
+    pub nr_cpuid_configs: u32,
+    pub cpuid_configs: [CpuidConfig; NR_CPUID_CONFIGS],
+}
+
+/// Provides information about the Intel TDX module
+#[derive(Debug)]
+pub struct TDXCapabilities {
+    pub fixed_attrs: FixedAttrs,
+    pub fixed_xfam: FixedXFAM,
+    /// supported Guest Physical Address Width
+    pub supported_gpaw: u32,
+    pub padding: u32,
+    /// set to 0
+    _reserved: [u64; 251],
+
+    pub cpuid_configuration: CpuidConfiguration,
+}
```

Comment:
Keep in mind this struct is what will be displayed to a user. Therefore, there's no reason we should include the `reserved` or `padding` space from the original kernel struct in it's Rust counterpart. Also, instead of a fixed array of CPUID configs (like that are present in the kernel) we can just present them as a `Vec<CpuidConfig>` in Rust.

```suggestion
/// Provides information about the Intel TDX module
#[derive(Debug)]
pub struct TDXCapabilities {
    pub fixed_attrs: FixedAttrs,
    pub fixed_xfam: FixedXFAM,
    /// supported Guest Physical Address Width
    pub supported_gpaw: u32,
    
    pub cpuid_configuration: Vec<CpuidConfig>,
}
```